pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/rule/security/ApexOpenRedirectRule.java

```diff
@@ -59,19 +71,19 @@ private void findSafeLiterals(AbstractApexNode<?> node) {
         }
     }
 
-    @Override
-    public Object visit(ASTFieldDeclaration node, Object data) {
-        findSafeLiterals(node);
-        return data;
-    }
-
     /**
      * Traverses all new declarations to find PageReferences
      * 
      * @param node
      * @param data
      */
     private void checkNewObjects(ASTNewObjectExpression node, Object data) {
+
+        ASTMethod method = node.getFirstParentOfType(ASTMethod.class);
+        if (method != null && Helper.isTestMethodOrClass(method)) {
```

Comment:
This code checks that the class is not marked with @isTest:
` public Object visit(ASTUserClass node, Object data) {
        if (Helper.isTestMethodOrClass(node)) {
            return data;
   }`

The code you highlighted checks that the method containing the new object expression is not a test. Also I have two separate test cases to support this code.